CAIPs/caip-x.md

```diff
@@ -0,0 +1,339 @@
+    "accounts": ["eip155:137:0xdef456"]
+  }
+}
+```
+
+**Example Request:**
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "provider_augment",
+  "params": {
+    "scopes": {
+      "eip155:1": {
+        "methods": ["eth_sendTransaction"],
+        "notifications": ["chainChanged"]
+      }
+    }
+  }
+}
+```
+
+**Example Response:**
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "success": true,
+    "updatedScopes": {
+      "eip155:1": {
+        "methods": ["eth_signTransaction", "eth_sendTransaction"],
+        "notifications": ["accountsChanged", "chainChanged"],
+        "accounts": ["eip155:1:0xabc123"]
+      },
+      "eip155:137": {
+        "methods": ["eth_sendTransaction"],
+        "notifications": ["chainChanged"],
+        "accounts": ["eip155:137:0xdef456"]
+      }
+    }
+  }
+}
+```
+
+**Explanation:**
+
+- The `provider_augment` method adds the specified scopes to the current session's permissions. If the scope already exists, the new methods and notifications are merged with the existing ones.
+
+#### `provider_revoke`
+
+Revokes permissions from an existing session.
+
+**Parameters:**
+
+- `sessionId` (string, optional): The session identifier.
+- `scopes` (object, required): An object containing the scopes to be revoked, formatted according to CAIP-217.
+
+**Initial Session Scopes:**
+
+```json
+{
+  "eip155:1": {
+    "methods": ["eth_signTransaction", "eth_sendTransaction"],
+    "notifications": ["accountsChanged", "chainChanged"],
+    "accounts": ["eip155:1:0xabc123", "eip155:1:0xdef456"]
+  },
+  "eip155:137": {
+    "methods": ["eth_sendTransaction"],
+    "notifications": ["chainChanged"],
+    "accounts": ["eip155:137:0xdef456"]
+  }
+}
+```
+
+**Example Request:**
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "provider_revoke",
+  "params": {
+    "scopes": {
+      "eip155:1": {
+        "methods": ["eth_sendTransaction"],
+        "notifications": ["chainChanged"],
+        "accounts": ["eip155:1:0xabc123"]
+      }
+    }
+  }
+}
+```
+
+**Example Response:**
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "success": true,
+    "updatedScopes": {
+      "eip155:1": {
+        "methods": ["eth_signTransaction"],
+        "notifications": ["accountsChanged"],
+        "accounts": ["eip155:1:0xdef456"]
+      },
+      "eip155:137": {
+        "methods": ["eth_sendTransaction"],
+        "notifications": ["chainChanged"],
+        "accounts": ["eip155:137:0xdef456"]
+      }
+    }
+  }
+}
+```
+
+**Explanation:**
+
+- The `provider_revoke` method removes the specified methods, notifications, and accounts from the current session's permissions. If the scope no longer has any methods, notifications, or accounts after revocation, it is removed from the session. If no scopes remain in the session, the session is considered closed. If no accounts remain in a scope, and only write methods the entire scope is removed.
```

Comment:
Either app or wallet can call this?
```suggestion
- The `provider_revoke` method removes the specified methods, notifications, and accounts from the current session's permissions. If a given `scopeObject` no longer has any methods, notifications, or accounts after executing this revocation, it is removed from the `sessionScopes` array. If no `scopeObject`s remain in the session after applying the above logic, the entire session is considered closed. If no accounts remain in a `scopeObject`, and no methods or only methods that would require an account to be called such as write methods, then the entire scope MUST be removed from the `sessionScopes` array of both wallet and application.
```

what does "considered closed" look like, behaviorally? Close the RPC session? Require a new _authorize call?  Reject all _augment calls until there is a new _authorize call?

Reply:
> Either app or wallet can call this?

Only dapp would call this.